// Copyright 2025 Ian Lewis
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package fx.v1;

import "google/type/date.proto";


// Currency represents a monetary currency as defined in ISO 4217.
message Currency {
    // alphabetic_code is the 3-letter currency code.
    string alphabetic_code = 1;

    // numeric_code is the currency's numeric code. The numeric code may be
    // zero for some historic currencies.
    string numeric_code = 2;

    // name is the human readable name for the currency.
    string name = 3;

    // minor_units is the decimal relationship between the currency's major and
    // minor units. "1", "2" and "3" signify a ratio of 10:1, 100:1 and 1000:1
    // respectively. If no minor units exist, this value is zero. Minor units
    // is zero for historic currencies.
    int32 minor_units = 4;

    // countries are the registered countries where the currency is/was used.
    repeated string countries = 5;

    // withdrawal_date is the historical date that the currency was withdrawn.
    // withdrawal_date.day can be be zero to indicate a month withdrawal date.
    google.type.Date withdrawal_date = 6;
}

// CurrencyList represents a list of Currency objects.
message CurrencyList {
    repeated Currency currencies = 1;
}
